tests/sequence/test_profile.py

```diff
@@ -0,0 +1,56 @@
+# This source code is part of the Biotite package and is distributed
+# under the 3-Clause BSD License. Please see 'LICENSE.rst' for further
+# information.
+
+import numpy as np
+import pytest
+import biotite.sequence as seq
+import biotite.sequence.align as align
+import biotite.sequence.profile as prof
+
+
+def test_from_alignment():
+    seq1 = seq.NucleotideSequence("CGTCAT")
+    seq2 = seq.NucleotideSequence("TCATGC")
+    ali_str = ["CGTCAT--",
+               "--TCATGC"]
+    trace = align.Alignment.trace_from_strings(ali_str)
+    alignment = align.Alignment([seq1, seq2], trace, None)
+
+    profile = prof.SequenceProfile.from_alignment(alignment)
+    position_frequency_matrix = np.array([[0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 2], [0, 2, 0, 0],
+                                          [2, 0, 0, 0], [0, 0, 0, 2], [0, 0, 1, 0], [0, 1, 0, 0]])
+    position_gaps = np.array([1, 1, 0, 0, 0, 0, 1, 1])
+    alphabet = seq.Alphabet(["A", "C", "G", "T"])
+    assert np.array_equal(position_frequency_matrix, profile.position_frequency_matrix)
+    assert np.array_equal(position_gaps, profile.position_gaps)
+    assert (alphabet == profile.alphabet)
+
+
+def test_to_consensus_nuc():
+    position_frequency_matrix = np.array([[0, 1, 0, 0], [0, 0, 1, 0], [0, 0, 0, 2], [0, 2, 0, 0],
+                                          [2, 0, 0, 0], [0, 0, 0, 2], [0, 0, 1, 0], [0, 1, 0, 0]])
+    position_gaps = np.array([1, 1, 0, 0, 0, 0, 1, 1])
+    alphabet = seq.Alphabet(["A", "C", "G", "T"])
+    profile = prof.SequenceProfile(position_frequency_matrix, position_gaps, alphabet)
+
+    assert "CGTCATGC" == profile.to_consensus()
```

Comment:
For completeness this function should also test the case where the resulting sequence contains an ambiguous symbol.